src/shared/journal-file-util.c

```diff
@@ -435,6 +435,14 @@ int journal_file_rotate(
         assert(f);
         assert(*f);
 
+#if HAVE_GCRYPT
+        /* Write the final tag */
+        if (JOURNAL_HEADER_SEALED((*f)->header) && journal_file_writable(*f)) {
+                r = journal_file_append_tag(*f);
+                if (r < 0)
+                        log_error_errno(r, "Failed to append tag when closing journal: %m");
+        }
+#endif
```

Comment:
this code exists twice now. 

I think this would ideally be a small helper call:

```
int journal_file_write_final_tag(JsonFile *f) {
        assert(f);
#if HAVE_GCRYPT
        if (!JOURNAL_HEADER_SEALED(f->header) || !journal_file_writable(f))
                return 0;

        r = journal_file_append_tag(f);
        if (r < 0)
                return log_debug_errno(r, "Failed to append tag when closing journal: %m");
#endif
        return 0;
}
```

and then call this here and in server_archive_offline_user_journals()

Reply:
It actually existed three times (as this snipped already existed before this PR). I created the helper and used it in all three places.